### PR TITLE
[Libtasn1] disable documentation

### PR DESCRIPTION
**Pull request by xandox**
Documentation generates by gtk tool which not must exists. For PR #15605

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  same, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by dg0yt** [comment]
This PR changes options to `configure` which default to being off anyway AFAIK. We have seen this package fail at least on osx already before `configure` due to failed attempts to run `gtkdocize` during `autoreconf` (#17098). If this PR runs into this problem, too:

I tried to get this solved for CI via provisioning (), and I made other proposals to solve this also for vcpkg users. The last attempt, #17406, doesn't seem to work as well as I thought for other packages, but that approach might perhaps be turned around in the libtasn1's portfile by adding
~~~
set(ENV{GTKDOCIZE} true)
~~~
before calling `vcpkg_configure_make`. (I cannot test this at the moment.)

**Comment by xandox** [comment]
`gt-html` is on by default according to `configure --help`

**Comment by dg0yt** [comment]
There are many changes to formatting. Some are improvements but aren't IMO.

File: ports/libtasn1/portfile.cmake [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
 set(VCPKG_CXX_FLAGS "-g -O2")
 
-vcpkg_configure_make(
-    AUTOCONFIG
-    SOURCE_PATH ${SOURCE_PATH}
-    OPTIONS
-        --disable-doc
-)
+vcpkg_configure_make(AUTOCONFIG SOURCE_PATH ${SOURCE_PATH} OPTIONS
+                     --disable-doc --disable-gtk-doc-html)
```
Comment (PhoebeHui) [inline ports/libtasn1/portfile.cmake:22]: Could you fix the format and indention and only include your changes here? We'd like to keep same format as before.

File: ports/libtasn1/portfile.cmake [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+  b356249535d5d592f9b59de39d21e26dd0f3f00ea47c9cef292cdd878042ea41ecbb7c8d2f02ac5839f5210092fe92a25acd343260ddf644887b031b167c2e71
 )
 
-vcpkg_extract_source_archive_ex(
-   OUT_SOURCE_PATH SOURCE_PATH
-   ARCHIVE ${ARCHIVE} 
-   REF ${VERSION}
-)
+vcpkg_extract_source_archive_ex(OUT_SOURCE_PATH SOURCE_PATH ARCHIVE ${ARCHIVE}
+                                REF ${VERSION})
```
Comment (PhoebeHui) [inline ports/libtasn1/portfile.cmake:14]: Could you fix the format here? We'd like to keep same format as before.

File: ports/libtasn1/portfile.cmake [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
-    FILENAME "libtasn1-${VERSION}.tar.gz"
-    SHA512 b356249535d5d592f9b59de39d21e26dd0f3f00ea47c9cef292cdd878042ea41ecbb7c8d2f02ac5839f5210092fe92a25acd343260ddf644887b031b167c2e71
+vcpkg_download_distfile(
+  ARCHIVE
+  URLS
+  "https://ftp.gnu.org/gnu/libtasn1/libtasn1-${VERSION}.tar.gz"
+  FILENAME
+  "libtasn1-${VERSION}.tar.gz"
+  SHA512
+  b356249535d5d592f9b59de39d21e26dd0f3f00ea47c9cef292cdd878042ea41ecbb7c8d2f02ac5839f5210092fe92a25acd343260ddf644887b031b167c2e71
```
Comment (PhoebeHui) [inline ports/libtasn1/portfile.cmake:10]: Could you fix the format and indention here? We'd like to keep same format as before.

File: ports/libtasn1/portfile.cmake [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
-vcpkg_fixup_pkgconfig() 
+vcpkg_fixup_pkgconfig()
 vcpkg_copy_pdbs()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(
+  INSTALL ${SOURCE_PATH}/LICENSE
+  DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}
+  RENAME copyright)
```
Comment (PhoebeHui) [inline ports/libtasn1/portfile.cmake:31]: Could you fix the format here? We'd like to keep same format as before.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@xandox, thanks for the PR!

**Comment by xandox** [comment]
Formatting is fixed. Sorry for that, did not notice my autoformat job.

**Comment by PhoebeHui** [comment]
No problem, thanks for your contribution!

**Review by PhoebeHui — APPROVED** [review]

**Comment by PhoebeHui** [comment]
The CI failed due to https://github.com/microsoft/vcpkg/issues/17057.

**Comment by ras0219-msft** [comment]
This LGTM, thanks for the improvement!

**Pull request merged**